org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -105,7 +108,18 @@ private void importDir(Path rootFolder, IProgressMonitor monitor) {
 
 	public static GradleDistribution getGradleDistribution(Path rootFolder) {
 		if (JavaLanguageServerPlugin.getPreferencesManager() != null && JavaLanguageServerPlugin.getPreferencesManager().getPreferences().isGradleWrapperEnabled() && Files.exists(rootFolder.resolve("gradlew"))) {
-			return GradleDistribution.fromBuild();
+			WrapperValidator validator = new WrapperValidator();
+			try {
+				String result = validator.checkWrapper(rootFolder.toFile().getAbsolutePath());
+				if (result == null) {
+					WrapperGradleDistribution gradleDistribution = GradleDistribution.fromBuild();
+					return gradleDistribution;
+				} else {
```

Comment:
A meaningful error message should eventually bubble up to the client. Listing all suspicious wrappers (if we find several)

Reply:
Fixed. We use/check only jars included to WrapperGradleDistribution.